resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -293,6 +293,9 @@ private[spark] class ApplicationMaster(args: ApplicationMasterArguments) extends
         }
 
         if (!unregistered) {
+          logInfo("Waiting for " + sparkConf.get("spark.yarn.report.interval", "1000").toInt +"ms to unregister am," +
```

Comment:
Use interpolation. No need to call `.toInt`. am -> AM or Application master. msg -> message; these should be complete sentences. You get the config twice here. Is this the default used elsewhere?

I don't know this code well and don't think a Thread.sleep is a great way to coordinate.

Reply:
This should also be a config constant. Instead of sleeping might be better to join `userClassThread` or `reporterThread` since they may exit more quickly than the configured wait.